Cargo.lock

```diff
@@ -0,0 +1,7 @@
+# This file is automatically @generated by Cargo.
```

Comment:
Why commit the lock file? Isn't it only recommended practice for applications and much less so for libraries? Just seems like a source of unnecessary churn for no benefit down the line, but I may be missing something.